contracts/staking/PoSValidatorManager.sol

```diff
@@ -425,29 +470,29 @@ abstract contract PoSValidatorManager is
         // a weight update using a higher nonce (which implicitly includes the delegation's weight update)
         // to be used to complete delisting for an earlier delegation. This is necessary because the P-Chain
         // is only willing to sign the latest weight update.
-        require(
-            $._delegatorStakes[delegationID].endingNonce <= nonce,
-            "PoSValidatorManager: nonce does not match"
-        );
+        require(delegator.endingNonce <= nonce, "PoSValidatorManager: nonce does not match");
 
         // Ensure the delegator is pending removed. Since anybody can call this function once
         // end delegation has been initialized, we need to make sure that this function is only
         // callable after that has been done.
         require(
-            $._delegatorStakes[delegationID].status == DelegatorStatus.PendingRemoved,
+            delegator.status == DelegatorStatus.PendingRemoved,
             "PoSValidatorManager: delegation not pending added"
         );
 
         // Update the delegator status
         $._delegatorStakes[delegationID].status = DelegatorStatus.Completed;
 
-        Delegator memory delegator = $._delegatorStakes[delegationID];
-        _unlock(delegator.weight, delegator.owner);
-        // TODO: issue rewards
+        _reward(delegator.owner, $._redeemableDelegatorRewards[delegationID]);
+        delete $._redeemableDelegatorRewards[delegationID];
+        _unlock(delegator.owner, weightToValue(delegator.weight));
+        // TODO can we remove the delegation from _delegatorStakes here?
```

Comment:
It seems like the answer here is yes. Once a delegationID's status is `Completed`, it shouldn't be able to be modified at all in the future right?